fdbservice/FDBService.cpp

```diff
@@ -643,37 +643,33 @@ class FDBService : public CServiceBase {
 		return true;
 	}
 
-	const char* getValueMultiUtil(const CSimpleIni& ini, const char* name, ...) {
+	const char* getValueMulti(const CSimpleIni& ini, const char* name, ...) {
 		const char* ret = nullptr;
 		const char* section = nullptr;
 		va_list ap;
 		va_start(ap, name);
 		while (!ret && (section = va_arg(ap, const char*))) {
 			ret = ini.GetValue(section, name, nullptr);
 		}
-		va_end(ap);
-		return ret;
-	}
-
-	const char* getValueMulti(const CSimpleIni& ini, const char* name, ...) {
-		va_list args;
-		va_start(args, name);
-		const char* ret = nullptr;
-		ret = getValueMultiUtil(ini, name, args);
 		if (!ret) {
-			std::string nameCopy(name);
-			for (int i = nameCopy.size() - 1; i >= 0; --i) {
-				if (nameCopy[i] == '-') {
-					nameCopy.at(i) = '_';
+			std::string keyCopy(key);
+			for (int i = keyCopy.size() - 1; i >= 0; --i) {
+				if (keyCopy[i] == '-') {
+					keyCopy.at(i) = '_';
 				}
 			}
-			ret = getValueMultiUtil(ini, name, args);
+			while (!ret && (section = va_arg(ap, const char*))) {
```

Comment:
I'm not sure this correctly uses `va_arg`, you may want to take a look at the documentation to get a sense for how to use these functions.

One option you could choose is to reduce this back down to one loop but check for both alternatives in the same loop.

Reply:
Technically, this still would only support exclusively hyphens or exclusively underscores rather than any combination of them like the other places we've made changes. I'm not sure if that's a showstopper though.

Reply:
This piece of code is definitely wrong. I have fixed it. Could you refresh and check it again?
Sorry for the late update